search-commons/src/main/java/no/unit/nva/search/constants/ApplicationConstants.java

```diff
@@ -63,4 +70,37 @@ private static String readSearchInfrastructureApiUri() {
     private static String readSearchInfrastructureAuthUri() {
         return ENVIRONMENT.readEnv("SEARCH_INFRASTRUCTURE_AUTH_URI");
     }
+
+
+    private static TermsAggregationBuilder generateSimpleAggregation(String term, String field) {
+        return AggregationBuilders
+                .terms(term)
+                .field(field)
+                .size(100);
+    }
+
+    private static NestedAggregationBuilder generateObjectLabelsAggregation(String object) {
+        return new NestedAggregationBuilder(object, object)
+                .subAggregation(generateIdAggregation(object));
+    }
+
+    private static TermsAggregationBuilder generateIdAggregation(String object) {
+        return new TermsAggregationBuilder(ID)
+                .field(jsonPath(object, ID))
+                .size(100)
+                .subAggregation(generateLabelsAggregation(object));
+    }
+
+    private static NestedAggregationBuilder generateLabelsAggregation(String object) {
+        return new NestedAggregationBuilder(LABELS, jsonPath(object, LABELS))
+                .subAggregation(generateSimpleAggregation(ENGLISH,  jsonPath(object, LABELS, ENGLISH, KEYWORD)))
+                .subAggregation(generateSimpleAggregation(BOKMAAL, jsonPath(object, LABELS, BOKMAAL, KEYWORD)))
+                .subAggregation(generateSimpleAggregation(NYNORSK, jsonPath(object, LABELS, NYNORSK, KEYWORD)))
+                .subAggregation(generateSimpleAggregation(SAMI, jsonPath(object, LABELS, SAMI, KEYWORD)));
+    }
```

Comment:
I feel like ApplicationConstants have grown out of its intented purpose. Should we at least move the 4 AggregationBuilder methods to a new AggregationBuilderUtils class?

Reply:
I'm going to leave this for a new PR since I will have to work more on the constants and I'm not sure what the scope of these changes will be yet